modules/criteoBidAdapter.js

```diff
@@ -31,12 +32,108 @@ const PUBLISHER_TAG_URL_TEMPLATE = 'https://static.criteo.net/js/ld/publishertag
 const FAST_BID_PUBKEY_E = 65537;
 const FAST_BID_PUBKEY_N = 'ztQYwCE5BU7T9CDM5he6rKoabstXRmkzx54zFPZkWbK530dwtLBDeaWBMxHBUT55CYyboR/EZ4efghPi3CoNGfGWezpjko9P6p2EwGArtHEeS4slhu/SpSIFMjG6fdrpRoNuIAMhq1Z+Pr/+HOd1pThFKeGFr2/NhtAg+TXAzaU=';
 
+const SID_COOKIE_NAME = 'cto_sid';
+const IDCPY_COOKIE_NAME = 'cto_idcpy';
+const LWID_COOKIE_NAME = 'cto_lwid';
+const OPTOUT_COOKIE_NAME = 'cto_optout';
+const BUNDLE_COOKIE_NAME = 'cto_bundle';
+const GUID_RETENTION_TIME_HOUR = 24 * 30 * 13; // 13 months
+const OPTOUT_RETENTION_TIME_HOUR = 5 * 12 * 30 * 24; // 5 years
+
 /** @type {BidderSpec} */
 export const spec = {
   code: BIDDER_CODE,
   gvlid: GVLID,
   supportedMediaTypes: [BANNER, VIDEO, NATIVE],
 
+  getUserSyncs: function (syncOptions, _, gdprConsent, uspConsent) {
+    const fastBidVersion = config.getConfig('criteo.fastBidVersion');
+    if (canFastBid(fastBidVersion)) {
+      return [];
+    }
+
+    const refererInfo = getRefererInfo();
+    const origin = 'criteoPrebidAdapter';
+
+    if (syncOptions.iframeEnabled && hasPurpose1Consent(gdprConsent)) {
```

Comment:
where is iframeEnabled coming from 

https://github.com/prebid/Prebid.js/blob/0be4560e182e46c121c283f763cdef74b5ff9db9/modules/pubmaticBidAdapter.js#L1300 is a more typical use of iframe syncs

Reply:
i see, you are trying to decorate the sync call with these id's.

Can you please indicate in your docs that 'opting into iframe syncs with this bidder may set values in the first party context'

Reply:
Regarding the _hasPurpose1Consent_ call
We really want the purpose1 to be granted prior to init our user sync as it involves interacting with 1PC the LS to resolve our sync endpoint arguments.